griffin/secret_management_util.py

```diff
@@ -78,15 +79,33 @@ def get_secret_from_vault(self, secret_key):
 
     def get_secret_from_cache(self, secret_key):
         cache_key = construct_cache_key(secret_key)
-        cache_value = self.redis_connection.get(cache_key)
+        cache_value = self.cache.get(cache_key)
         if cache_value is None:
             raise SecretNotFoundException(f"Secret Unavailable in cache for secret key - {secret_key}")
-        return cache_value.decode("utf-8")
+        LOGGER.info(f'Secret Value fetched from cache for secret key {secret_key}')
+        return cache_value
 
     def set_secret_to_cache(self, key, value):
         cache_key = construct_cache_key(key)
         try:
-            self.redis_connection.set(cache_key, value)
-            self.redis_connection.expire(cache_key, GRIFFIN_CACHE_TTL)
+            self.cache[cache_key] = value
         except Exception as e:
-            LOGGER.error(f"Failed to set cache value for cache key {cache_key} - {str(e)}")
\ No newline at end of file
+            LOGGER.error(f"Failed to set cache value for cache key {cache_key} - {str(e)}")
+        LOGGER.info(f"Successfully set cache for secret key {key}")
+
+    def warm_up_cache(self, secret_keys):
+        try:
+            vault_client = self.vault_client()
+            for secret_key in secret_keys:
+                keypath = get_keypath_from_input(secret_key)
```

Comment:
@suprithak16 can we check if the key already exists in  the local cache, if yes, then we dont need to pull

Reply:
also, this looks like duplicate code, can we not use get_secret_from_vault here?

Reply:
1. is done 2. yes the duplicate was known , it was done that way because get_secret_from_vault had a line of code that would establish a connection to the vault, and this would be done for all secrets. Have moved that logic to init and reused get_secret_from_vault now please check